### PR TITLE
Access-Control: Document changes to the provisioner

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:
We slightly update the behavior of the access-control provisioner after the 8.0 release.
1. Deletion happens before insertion.
2. Assignments are updated when version remains unchanged.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1565

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/access-control/provisioning.md [hunk 1]
```diff
@@ -92,6 +94,8 @@ deleteRoles:
 
 To assign roles to built-in roles, add said built-in roles to the `builtInRoles` section of your roles. To remove a specific assignment, remove it from the list.
 
+> Assignments are updated if the version of the role is greater or equal to the one stored in database. You don't need to increment the version number of the role to update its assignments.
```
Comment (gamab) [inline docs/sources/enterprise/access-control/provisioning.md:97]:
```suggestion
> Assignments are updated if the version of the role is greater or equal to the one stored internally. You don't need to increment the version number of the role to update its assignments.
```

**Comment by osg-grafana** [comment]
I removed the baked in release number in the title and should we add a milestone instead?

File: docs/sources/enterprise/access-control/provisioning.md [hunk 2]
```diff
@@ -74,7 +74,9 @@ The `orgId` is lost when the role is set to global.
 
 ### Delete roles 
 
-To delete a role, you can add a list of roles under the `deleteRoles` section in the configuration file. Such a deletion is performed after a role is inserted or updated.
+To delete a role, you can add a list of roles under the `deleteRoles` section in the configuration file.
```
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/provisioning.md:77]:
```suggestion
To delete a role, add a list of roles under the `deleteRoles` section in the configuration file. 
```

File: docs/sources/enterprise/access-control/provisioning.md [hunk 3]
```diff
@@ -74,7 +74,9 @@ The `orgId` is lost when the role is set to global.
 
 ### Delete roles 
 
-To delete a role, you can add a list of roles under the `deleteRoles` section in the configuration file. Such a deletion is performed after a role is inserted or updated.
+To delete a role, you can add a list of roles under the `deleteRoles` section in the configuration file. 
+
+> Deletion of roles listed under the `deleteRoles` section happens **before** saving roles listed under the `roles` section.
```
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/provisioning.md:79]:
```suggestion
> **Note:** Any role in the `deleteRoles` section is deleted before any role in the `roles` section is saved.
```

**Comment by gamab** [comment]
@osg-grafana  thanks! I added the 8.1 milestone :)

File: docs/sources/enterprise/access-control/provisioning.md [hunk 4]
```diff
@@ -92,6 +94,8 @@ deleteRoles:
 
 To assign roles to built-in roles, add said built-in roles to the `builtInRoles` section of your roles. To remove a specific assignment, remove it from the list.
 
+> Assignments are updated if the version of the role is greater or equal to the one stored internally. You don't need to increment the version number of the role to update its assignments.
```
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/provisioning.md:97]:
```suggestion
> **Note:** Assignments are updated if the version of the role is greater or equal to the one stored internally. You don’t need to increment the version number of the role to update its assignments.
```

**Review by osg-grafana — APPROVED** [review]

**Review by osg-grafana — APPROVED** [review]

**Pull request merged**